app/models/course/assessment/answer/programming.rb

```diff
@@ -58,7 +58,25 @@ def assign_params(params)
 
     params[:files_attributes]&.each do |file_attributes|
       file = files.find { |f| f.id == file_attributes[:id].to_i }
-      file.content = file_attributes[:content]
+      file.content = file_attributes[:content] if file.present?
     end
   end
+
+  def create_and_update_files(params)
+    params[:files_attributes]&.each do |file_attributes|
+      file = files.find { |f| f.id == file_attributes[:id].to_i }
+      if file.present?
+        file.content = file_attributes[:content]
+      else
+        files.build(filename: file_attributes[:filename], content: file_attributes[:content])
+      end
+    end
+    specific.save
```

Comment:
`save` will do, no need for specific since you are at the programming question model.